### PR TITLE
sql: change ordering of (re)creating secondary indexes in ALTER PK

### PR DESCRIPTION
**Pull request by Xiang-Gu**
If we alter the primary key on a table with secondary indexes, we will recreate those secondary indexes as well as creating a new unique secondary index on the old primary key column(s). Previously, in the legacy schema changer, we'd create the new unique secondary index first and then recreate all existing secondary indexes. This is different from how declarative schema changer processes ALTER PK, in which the ordering is the opposite. This is not an issue per se; the only time it shows a difference is when you inspect the table after altering PK via, say, `SHOW CREATE t`, where you'd then observe a slightly different ordering of the indexes. This is actually easier to change the legacy schema changer so both schema changer would produce the same ordering of secondary indexes after an ALTER PK. This will help clean several test cases.

Epic: None
Release note: None

### CODE REVIEW

**Comment by Xiang-Gu** [comment]
Another motivation to prompt this non-functional change is to facilitate WIP comparator testing where we rely on `SHOW CREATE` to compare two schema states.

File: pkg/sql/logictest/testdata/logic_test/alter_primary_key [hunk 1]
```diff
@@ -1397,23 +1397,6 @@ a  b  k
...
-CREATE TABLE public.t (
-  a INT8 NOT NULL,
-  b INT8 NOT NULL,
-  k INT8 NOT NULL AS (a + b) VIRTUAL,
-  CONSTRAINT t_pkey PRIMARY KEY (k ASC),
-  UNIQUE INDEX t_a_key (a ASC),
-  INDEX t_idx_b_k (b ASC, k ASC)
-)
-
-skipif config local-legacy-schema-changer
```
Comment (Xiang-Gu) [inline pkg/sql/logictest/testdata/logic_test/alter_primary_key:1416]: This is an example where previously this slight difference required us to write two expected output for the same thing (one for legacy schema changer and another for declarative schema changer). This is very ugly and I'm glad we can get rid of them now.

**Review by fqazi — APPROVED** [review]
:lgtm: Its a small but good find :)

> Reviewed 2 of 2 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110068#-:-NdfJN8r2Qx4Cdxcolhw:b431j3l)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by Xiang-Gu** [comment]
TFTR!
bors r+

**Pull request merged**